postgresql/config/functions.sh

```diff
@@ -67,10 +67,17 @@ bootstrap_replica_via_pg_basebackup() {
 }
 
 ensure_dir_ownership() {
-  echo 'Making sure hab user owns var and data paths'
-  chown -RL hab:hab {{pkg.svc_var_path}}
-  chown -RL hab:hab {{pkg.svc_data_path}}
-  chmod 0700 {{pkg.svc_data_path}}/pgdata
+  paths="{{pkg.svc_var_path}} {{pkg.svc_data_path}}"
+  if [[ $EUID -eq 0 ]]; then
+    # if EUID is root, so we should chown to pkg_svc_user:pkg_svc_group
+    ownership_command="chown -RL {{pkg.svc_user}}:{{pkg.svc_group}} $paths"
+  else
+    # not root, so at best we can only chgrp to the effective user's primary group
+    ownership_command="chgrp -RL $(id -g) $paths"
+  fi
+  echo "Ensuring proper ownership: $ownership_command"
+  $ownership_command  || true
```

Comment:
Why the `|| true` here?

Reply:
@elliott-davis it's not necessary, however it does provide a best effort fail safe for the edge case that we possibly haven't yet considered.

Reply:
So the intention behind it is to not fail the hook in the chance that your chown command fails? I think I would prefer to fail fast and know that we have a permission issue

Reply:
Let's do this! I'm removing the `|| true` and pushing it up. After additional reviews and approval I'll squash all commits.